config/runtime/tests/watchdog-reset.jinja2

```diff
@@ -0,0 +1,12 @@
+- test:
+    timeout:
+      minutes: {{ test_timeout|default('2') }}
+    interactive:
+    - name: wdt-reset
+      prompts: ['{{ rootfs_prompt }}']
+      script:
+      - name: wdt-expect-reset
+        command: "sleep {{ wdt_timeout|default('30') | int + 5 }} > {{ wdt_dev|default('/dev/watchdog')}}"
```

Comment:
A comment explaining how/why this command works would be nice (I assume the shell command is blocking the watchdog device file, so the OS can't reset it)

OOC, could the WDT timeout be retrieved from somewhere else, like a sysfs entry?

Reply:
This command opens the watchdog device and just waits for the timeout to expire, causing the device to reset. If the reset occurs as expected, a bootloader string should be matched ; if the command returns successfully without a reset, the prompt string will be matched instead. I can add a comment in the template itself to clarify this.

> OOC, could the WDT timeout be retrieved from somewhere else, like a sysfs entry?

Yes, according to [the docs](https://www.kernel.org/doc/Documentation/ABI/testing/sysfs-class-watchdog) the timeout is exposed by `/sys/class/watchdog/watchdogX/timeout`. Afaik there's no way to define variables in LAVA interactive tests, but we could change the command to `sleep $(( $(cat /sys/class/watchdog/watchdogX/timeout) + 2 )) > /dev/watchdog` - it's less readable, but the user would just need to provide the name of the watchdog device. I'll run a quick manual test to check if this even works when integrated in the LAVA definition..

Reply:
> This command opens the watchdog device and just waits for the timeout to expire, causing the device to reset.

Does that mean no other process is likely to reset the watchdog?

> Afaik there's no way to define variables in LAVA interactive tests

Too bad it doesn't accept a `commands:` param with a list of commands like we have in other tests...

> we could change the command to [...]

I'll trust your judgement here, FWIW this question was really driven by curiosity, I'd be fine with a comment explaining how this works so future devs don't need to scratch their head.

Reply:
> Does that mean no other process is likely to reset the watchdog?

This test assumes there's no other watchdog service running on the system - this applies to our current `bookworm` rootfs, so I didn't feel the need to create an ad-hoc rootfs for this test.

> Too bad it doesn't accept a commands: param with a list of commands like we have in other tests...

Actually, I realized it's possible to include more than one command and define a variable within that context. I force-pushed some changes - would you mind reviewing them and see if they look good to you?

I manually tested the changes and they seem to work as expected - e.g.: https://lava.collabora.dev/scheduler/job/14061709

Reply:
> This test assumes there's no other watchdog service running on the system - this applies to our current `bookworm` rootfs

Ah, that's worth mentioning in the comment as well.

> Actually, I realized it's possible to include more than one command and define a variable within that context.

Awesome, thanks for checking!